### PR TITLE
feat: added add note to trash, archive note and delete note permanently.

### PR DESCRIPTION
**Pull request by devdeadviz**
# Change Logs
- Add move note to trash functionality.
- Restore deleted note functionality.
- Permanently delete notes from trash functionality.
- Move to note archive functionality added.
- Restore archive note functionality added.
- Move archive note to trash functionality added.
- Added toasts.
## Deploy Preview Link: https://deploy-preview-5--devdeadviz-my-notes.netlify.app/

### CODE REVIEW

**Review by HarshithaKonduru214 — APPROVED** [review]
Things I liked:
1. Good use of barrel imports.
2. Components broken down nicely.
3. Clean code structure

File: src/reducers/noteReducer.js [hunk 1]
```diff
@@ -6,6 +6,20 @@ const noteReducer = (state, action) => {
       return { ...state, editedNotes: true };
     case "UPDATE_NOTE":
       return { ...state, newNotes: action.payload, editedNotes: false };
+    case "TRASH_NOTE":
+      return { ...state, trashNotes: [...state.trashNotes, action.payload] };
```
Comment (Shruthi-Babu) [inline src/reducers/noteReducer.js:10]: You can destructure action here.

**Review by Shruthi-Babu — APPROVED** [review]
Good folder structure
Good naming convention

File: src/reducers/noteReducer.js (hunk 1 above)
Comment (devdeadviz) [reply src/reducers/noteReducer.js:10]: Okay

**Pull request merged**